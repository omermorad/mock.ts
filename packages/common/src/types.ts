import faker from 'faker';
import FakerStatic = Faker.FakerStatic;

export type Faker = FakerStatic;
export const Faker = faker;

export interface ObjectLiteral {
  [key: string]: unknown;
}

export type Class<T = any> = new (...args: any[]) => T;

export type AbsoluteValue = string | number | boolean | ObjectLiteral | RegExp | Date;

export type FakerCallback = (faker: Faker) => any;

export type LazyType<T = any> = () => Class<T>;

export type Enum<E> = Record<keyof E, number | string> & { [key: number]: string };

export type LazyEnum<E = any> = { enum: () => Enum<E> };

export type ClassLiteral<TClass = any> = { [K in keyof TClass]: TClass[K] };

export type OptionalClassValues<TClass> = Partial<ClassLiteral<TClass>>;
