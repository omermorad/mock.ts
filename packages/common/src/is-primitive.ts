/**
 * Checks if a given constructor name is a primitive one
 *
 * @param constructorName
 * @returns {boolean}
 */
export function isPrimitive(constructorName: string): boolean {
  return ['String', 'Boolean', 'Number', 'Date'].includes(constructorName);
}
