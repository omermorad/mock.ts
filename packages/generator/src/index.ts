export * from '@mockinbird/parser';
export * from '@mockinbird/types';

export { Mock, MockOptions } from '@mockinbird/reflect';

export * from './lib';
export * from './types/mock-generator-options.interface';
