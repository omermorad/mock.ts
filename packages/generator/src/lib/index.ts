export * from './mock-factory';
export * from './builder';
