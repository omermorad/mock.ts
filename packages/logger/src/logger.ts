import chalk from 'chalk';
import util from 'util';

const prefix = '\u{1F426} ';

function format(args: Array<any>, customPrefix?: string) {
  const fullPrefix = prefix + (customPrefix === undefined ? '' : ' ' + customPrefix);
  return (
    fullPrefix +
    util
      .format('', ...args)
      .split('\n')
      .join('\n' + fullPrefix + ' ')
  );
}

export interface MockingbirdLogger {
  info(message: string, ...args: any[]): void;
  error(message: string, ...args: any[]): void;
  warn(message: string, ...args: any[]): void;
  log(message: string, ...args: any[]): void;
  success(message: string, ...args: any[]): void;
}

export const Logger: MockingbirdLogger = {
  info: (...args: any[]): void => {
    console.info(format(args, chalk.cyan('info')));
  },
  log: (...args: any[]): void => {
    console.log(format(args));
  },
  error: (...args: any[]): void => {
    console.error(format(args, chalk.red('error')));
  },
  success: (...args: any[]): void => {
    console.log(format(args, chalk.green('success')));
  },
  warn: (...args: any[]): void => {
    console.warn(format(args, chalk.yellow('warn')));
  },
};
