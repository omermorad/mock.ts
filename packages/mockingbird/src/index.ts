export { Mock, DecoratorValueArg } from '@mockingbird/reflect';

export * from '@mockingbird/parser';
export * from '@mockingbird/common';

export * from './lib';
