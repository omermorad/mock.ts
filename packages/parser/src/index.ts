export * from './lib/class-parser';
export * from './types/parser-config-dto';
