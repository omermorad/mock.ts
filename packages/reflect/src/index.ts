export * from './types';
export * from './decorators';
export * from './lib';
