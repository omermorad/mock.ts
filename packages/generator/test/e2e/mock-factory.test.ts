import { MockFactory } from '../../src';
import { TestClassesE2E } from './test-classes';
import Dog = TestClassesE2E.Dog;

describe('MockFactory e2e Test', () => {
  when("calling 'many' method with a given class", () => {
    let result: Dog[];

    beforeAll(() => {
      result = MockFactory<Dog>(Dog).many(3);
    });

    then('return an array with the exact same length', () => {
      expect(result).toBeInstanceOf(Array);
      expect(result).toHaveLength(3);
    });

    then('return 3 instances of the given class', () => {
      expect(result).toMatchSnapshot();
    });
  });
});
