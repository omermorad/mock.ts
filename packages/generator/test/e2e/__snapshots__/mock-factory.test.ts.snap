// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MockFactory e2e Test when calling 'many' method with a given class then return 3 instances of the given class 1`] = `
Array [
  Dog {
    "name": "doggy dog",
  },
  Dog {
    "name": "doggy dog",
  },
  Dog {
    "name": "doggy dog",
  },
]
`;
